aiida/engine/processes/process.py

```diff
@@ -635,8 +631,8 @@ def decode_input_args(self, encoded: str) -> Dict[str, Any]:  # pylint: disable=
         return serialize.deserialize_unsafe(encoded)
 
     def update_node_state(self, state: plumpy.process_states.State) -> None:
-        self.update_outputs()
         self.node.set_process_state(state.LABEL)
+        self.update_outputs()
```

Comment:
we can see if we can move this to something like `on_terminate` so as to only do it once. Not sure if this would break anything but it should be more performant. Will have a look tomorrow.